File: backend/src/main/java/com/example/backend/model/Student.java
package com.example.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Student {
    @Id
    @GeneratedValue
    private Long id;
    private Integer registrationNumber;
    private String firstName;
    private String lastName;
    private String localAddress;
    private String permanentAddress;

    public Student(Long id) {
        this.id = id;
    }
}

File: frontend/src/app/dto/Student.ts
import { Course } from "./Course";
import { Department } from "./Department";
import { User } from "./User";

export class Student {

  id: number | null = null;
  firstName: string | null = null;
  lastName: number | null = null;
  localAddress: string | null = null;
  permanentAddress: string | null = null;
  department: Department | null = null;
  course: Array<Course> | null = null;
  user: User | null = null;

  constructor(o?: Partial<Student>) {
    Object.assign(this, o);
  }
}
